### PR TITLE
[MISC] Doxygen Fixes Constructors in IO

### PR DESCRIPTION
**Pull request by MitraDarja**
To fix doxygen warning, part I.

### CODE REVIEW

File: include/seqan3/io/alignment_file/format_sam.hpp [hunk 1]
```diff
@@ -118,15 +118,17 @@ class alignment_file_format_sam
...
+    //!\brief Default constructor is explicitly defaulted, you need to give a stream or file name.
     alignment_file_format_sam() = default;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
     alignment_file_format_sam(alignment_file_format_sam const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
     alignment_file_format_sam & operator=(alignment_file_format_sam const &) = delete;
-    alignment_file_format_sam(alignment_file_format_sam &&) = default;
-    alignment_file_format_sam & operator=(alignment_file_format_sam &&) = default;
-    ~alignment_file_format_sam() = default;
+    alignment_file_format_sam(alignment_file_format_sam &&) = default; //!< Defaulted
```
Comment (smehringer) [inline include/seqan3/io/alignment_file/format_sam.hpp]: Can you align the comments? :grin: :nail_care: 
Also in all the other cases

File: include/seqan3/io/detail/ignore_output_iterator.hpp [hunk 2]
```diff
@@ -42,16 +42,16 @@ class ignore_output_iterator
     using iterator_category = std::output_iterator_tag;
     //!\}
 
-    /*!\name Constructor, destructor and assignment
+    /*!\name Constructors, destructor and assignment
+     * \brief Rule of five explicitly defaulted.
```
Comment (smehringer) [inline include/seqan3/io/detail/ignore_output_iterator.hpp]: You deleted this comment in the other cases so I would not introduce this line here

File: include/seqan3/io/alignment_file/format_sam.hpp [hunk 3]
```diff
@@ -118,15 +118,17 @@ class alignment_file_format_sam
 {
 public:
     /*!\name Constructors, destructor and assignment
-     * \brief Rule of five explicitly defaulted.
      * \{
      */
+    //!\brief Default constructor is explicitly defaulted, you need to give a stream or file name.
     alignment_file_format_sam() = default;
```
Comment (eseiler) [inline include/seqan3/io/alignment_file/format_sam.hpp]: Please delete the brief and do a `//!< Defaulted`

File: include/seqan3/io/sequence_file/format_fasta.hpp [hunk 4]
```diff
@@ -80,15 +80,17 @@ class sequence_file_format_fasta
 {
 public:
     /*!\name Constructors, destructor and assignment
-     * \brief Rule of five explicitly defaulted.
      * \{
      */
+    //!\brief Default constructor is explicitly defaulted, you need to give a stream or file name.
     sequence_file_format_fasta() = default;
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/format_fasta.hpp]: same here

File: include/seqan3/io/sequence_file/format_fastq.hpp [hunk 5]
```diff
@@ -77,14 +77,16 @@ class sequence_file_format_fastq
 {
 public:
     /*!\name Constructors, destructor and assignment
-     * \brief Rule of five explicitly defaulted.
      * \{
      */
+    //!\brief Default constructor is explicitly defaulted, you need to give a stream or file name.
     sequence_file_format_fastq() = default;
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/format_fastq.hpp]: and here

File: include/seqan3/io/structure_file/format_vienna.hpp [hunk 6]
```diff
@@ -85,14 +85,16 @@ class structure_file_format_vienna
 {
 public:
     /*!\name Constructors, destructor and assignment
-     * \brief Rule of five explicitly defaulted.
      * \{
      */
+    //!\brief Default constructor is explicitly defaulted, you need to give a stream or file name.
     structure_file_format_vienna() = default;
```
Comment (eseiler) [inline include/seqan3/io/structure_file/format_vienna.hpp]: also here

File: include/seqan3/io/structure_file/format_vienna.hpp [hunk 7]
```diff
@@ -85,14 +85,16 @@ class structure_file_format_vienna
...
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
     structure_file_format_vienna(structure_file_format_vienna const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
     structure_file_format_vienna & operator=(structure_file_format_vienna const &) = delete;
-    structure_file_format_vienna(structure_file_format_vienna &&) = default;
-    structure_file_format_vienna & operator=(structure_file_format_vienna &&) = default;
+    structure_file_format_vienna(structure_file_format_vienna &&) = default;             //!< Defaulted
+    structure_file_format_vienna & operator=(structure_file_format_vienna &&) = default; //!< Defaulted
```
Comment (eseiler) [inline include/seqan3/io/structure_file/format_vienna.hpp]: Can you please add 
```
~ structure_file_format_vienna() = default; //!< Defaulted
```

The destructor is somehow missing here...

File: include/seqan3/io/sequence_file/format_fastq.hpp [hunk 8]
```diff
@@ -77,14 +77,16 @@ class sequence_file_format_fastq
...
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
     sequence_file_format_fastq(sequence_file_format_fastq const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
     sequence_file_format_fastq & operator=(sequence_file_format_fastq const &) = delete;
-    sequence_file_format_fastq(sequence_file_format_fastq &&) = default;
-    sequence_file_format_fastq & operator=(sequence_file_format_fastq &&) = default;
+    sequence_file_format_fastq(sequence_file_format_fastq &&) = default;             //!< Defaulted
+    sequence_file_format_fastq & operator=(sequence_file_format_fastq &&) = default; //!< Defaulted
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/format_fastq.hpp]: Please add the destructor (see sequence file format comment)

File: include/seqan3/io/record.hpp [hunk 9]
```diff
@@ -164,14 +164,13 @@ struct record : detail::transfer_template_args_onto_t<field_types, std::tuple>
...
-    record() = default;
-    record(record const &) = default;
-    record & operator=(record const &) = default;
-    record(record &&) = default;
-    record & operator=(record &&) = default;
+    record() = default;                           //!< Defaulted
+    record(record const &) = default;             //!< Defaulted
+    record & operator=(record const &) = default; //!< Defaulted
+    record(record &&) = default;                  //!< Defaulted
+    record & operator=(record &&) = default;      //!< Defaulted
```
Comment (eseiler) [inline include/seqan3/io/record.hpp]: add destructor

**Pull request merged**